cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/RecoverFromStorageCommand.java

```diff
@@ -0,0 +1,595 @@
+        }
+    }
+
+    /**
+     * Method to set the highest epoch seen while parsing metadata chunks.
+     * @param chunk Chunk to extract epoch from.
+     * @param containerId Container ID of the the container being recovered.
+     * @param containersToEpoch Map to hold highest epoch for all containers.
+     */
+    private void setEpochforContainer(File chunk, int containerId, Map<Integer, Long> containersToEpoch) {
+        long epoch = getEpochFromChunk(chunk);
+        long epochAlreadyPresent = containersToEpoch.getOrDefault(containerId, Long.valueOf(1));
+        if ( epoch > epochAlreadyPresent ) {
+            containersToEpoch.put(containerId, epoch);
+        }
+    }
+
+    /**
+     * Extracts epoch from chunk name.
+     */
+    private long getEpochFromChunk(File chunk) {
+        String[] chunkParts = chunk.getName().split(EPOCH_SPLITTER);
+        return Long.parseLong(chunkParts[chunkParts.length - 1].split("-")[0]);
+    }
+
+    private File[] getContainerMetadataChunkFiles(File[] allFiles, int containerId) {
+        List<File> filteredFiles =  Arrays.stream(allFiles)
+                .filter(File::isFile)
+                .filter(f -> !f.getName().contains(ATTRIBUTE_SUFFIX))
+                .filter(f -> f.getName().split(EPOCH_SPLITTER)[0].contains("_" + String.valueOf(containerId)))
+                .collect(Collectors.toList());
+        return filteredFiles.toArray(new File[filteredFiles.size()]);
+    }
+
+    private void deleteJournalsOf(int containerId) {
+        File dir = new File(tier2Root + containersPath);
+        assert dir != null;
+        File[] files = dir.listFiles();
+        assert files != null;
+        for (File file : files) {
+            if (file.getName().contains("container" + String.valueOf(containerId))) {
+                file.delete();
+            }
+        }
+    }
+
+    /**
+     * Reconciles  all segment metadata in a container, taking Storage_Metadata
+     * as source of truth.
+     * @param container
+     * @throws Exception
+     */
+    private void reconcileStorageSegment(DebugStreamSegmentContainer container) throws Exception {
+        Map<Integer, Set<String>> segmentsByContainer = ContainerRecoveryUtils.getExistingSegments(Map.of(container.getId(), container), executorService, true, TIMEOUT );
+            Set<String> segments = segmentsByContainer.get(container.getId());
+        ContainerTableExtension extension = container.getExtension(ContainerTableExtension.class);
+
+        for (String segment : segments) {
+            List<TableEntry> entries = extension.get(NameUtils.getStorageMetadataSegmentName(container.getId()), Collections.singletonList(BufferView.wrap(segment.getBytes(StandardCharsets.UTF_8))), TIMEOUT).get();
+            TableEntry entry = entries.get(0);
+            StorageMetadata storageMetadata = SLTS_SERIALIZER.deserialize(entry.getValue().getCopy()).getValue();
```

Comment:
We need to check that segment metadata is marked active and exclude segments marked for deletion,

https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/SegmentMetadata.java#L185-L192

Reply:
updated to check that the segment is marked active now. 
Will test excluding segments marked for deletion, more before commiting the change.